File: src/main/java/com/example/quanlynhansu/service/WorkReportService.java
package com.example.quanlynhansu.service;

import com.example.quanlynhansu.entity.WorkReport;

import java.util.List;

public interface WorkReportService {
    WorkReport create(WorkReport report);
    WorkReport update(Long id, WorkReport report);
    void delete(Long id);
    WorkReport getById(Long id);
    List<WorkReport> getAll();
    List<WorkReport> getByUserId(Long userId);
}

File: forntend_quanlynhansu/src/services/WorkReportService.js
import api from "./BaseApi";

const WorkReportService = {
    getAll: () => api.get("/work-reports"),
    getById: (id) => api.get(`/work-reports/${id}`),
    create: (data) => api.post("/work-reports", data),
    update: (id, data) => api.put(`/work-reports/${id}`, data),
    delete: (id) => api.delete(`/work-reports/${id}`),
    getByUserId: (userId)=>api.get(`/work-reports/user/${userId}`),
};

export default WorkReportService;
